pkg/sensors/tracing/genericlsm.go

```diff
@@ -239,6 +239,12 @@ func addLsm(f *v1alpha1.LsmHookSpec, in *addLsmIn) (id idtable.EntryID, err erro
 		argType := gt.GenericTypeFromString(a.Type)
 
 		if a.Resolve != "" && j < api.EventConfigMaxArgs {
+			if !kernels.MinKernelVersion("5.8.0") {
+				// bpf_lsm_<hook_name> does not exist in BTF file before 5.8
+				// https://lore.kernel.org/bpf/20200329004356.27286-4-kpsingh@chromium.org/
+				return errFn(fmt.Errorf("Error: LSM programs could not use Resolve flag for your kernel version." +
+					"Please use Kprobe instead or update your kernel to version 5.8 or higher"))
+			}
```

Comment:
Checking kernel versions does not always work as expected because there are kernels with older versions that backport a lot of functionality.

Am I correct to assume that the `resolveBtfArg` would fail anyway? If so, I would suggest something like:
```
			lastBtfType, btfArg, err := resolveBtfArg("bpf_lsm_"+f.Hook, a)
                        if err != nil {
                                        return errFn(fmt.Errorf("could not resolve  path from btf. You can try to use a kprobe or update your kernel version to 5.8 or later"))
                        }
```

Reply:
One thing to note is that there are vendor kernels (e.g., rhel) that do not follow kernel.org kernels. So a kernel might report it's 4.18, for example, but have a number of patches that are not part of the 4.18 kernel. This is why, in many cases, kernel versions do not tell the full story and it's better to test whether a feature is available or not.

I think an error message like:
> fmt.Errorf("failed to find BTF type for hook %q: %w. Please check if the hook exists or if your kernel supports BTF for lsm hooks. As an alternative, consider switching to kprobes. ", hook, err)

is perfectly find, but either approach works.